app/reports.py

```diff
@@ -52,7 +52,7 @@ class AQI:
     @staticmethod
     def get_quality(source):
         concentration = next(
-            c for c in CONCENTRATIONS if c[0] <= source.average <= c[1]
+            c for c in CONCENTRATIONS if c[0] <= source.stats <= c[1]
```

Comment:
You forgot to run the test :) this change is breaking everything, because source.average is the attribute with the average for that source, not the Measurements function.